### PR TITLE
Recipe App Wireframe

### PR DESCRIPTION
**Pull request by AKeeganDev**
## Setup

Once you have cloned the library run these commands in order:

- `rails db:create`
- `rails db:migrate`
- `rails db:seed`

## Using the program

For using the program for the first time we recommend that you use one of the users already created and verified through the seed. Here are the credentials and instructions to do that

1. Run `rails s` to start the server and navigate to localhost:3000
2. login using the following --- email: "augusto@dev.com" password: "123456"
3. From here you can navigate and use all of the features requested in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md)

## Features of this pull request

- Login page and registration page built with Devise
- Displays a list of food added by the logged-in user along with the quantity of the given food.
- Form that allows users to add new foods.
- Recipes list displays a list of recipes created by the logged-in user.
- Clicking a recipe leads to recipe details.
- If the user is the owner of the recipe then only user can delete it (unless the user is an Admin).
- Public recipe list displays a list of all public recipes.
- Clicking recipes in the public recipes page will lead to recipe details.
- Recipe details displays that the recipe is public and displays the recipe details.
- If the user is the owner of the recipe, this displays a form that allows the user to add new food.
- General shopping list view shows the list of foods that are missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- Counts the total food items and total price of the missing food.
-  Navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,70 @@
...
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Thanks to my peers who are always there to offer support.
+
+## 📝 License
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (MahmoudBakr23) [inline README.md:70]: **Optional**
- Please fix the MIT link for a professional README file.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (MahmoudBakr23) [inline spec/models/user_spec.rb]: - [x] I can see that you might have forgotten to write unit and integration tests for your app. Please install the RSpec configs and do so. 😄

File: README.md [hunk 3]
```diff
@@ -1,24 +1,70 @@
-# README
+# Recipe App
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+> The Recipe app keeps track of all your recipes, ingredients, and inventory. It will allow you to save ingredients, keep track of what you have, create recipes, and generate a shopping list based on what you have and what you are missing from a recipe. Also, since sharing recipes is an important part of cooking the app should allow you to make them public so anyone can access them.
 
-Things you may want to cover:
+## Getting Started
```
Comment (MahmoudBakr23) [inline README.md:5]: **Optional**
- It'd be awesome if you deployed your app and attach the live link in the README file

**Review by MahmoudBakr23 — APPROVED** [review]
Hello team😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪

- Well done setting up your migrations and schema correctly
- Great job creating the models and their associations and validations
- Great work with the controllers and the RESTful methods
- Well done applying the given design in the views using SCSS
- Well done specifying the routes
- Great work with devise gem
- Well done adding partials for the views
- Descriptive PR

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
_See the comments down below._

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention me @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (jsug9) [reply README.md:70]: Fixed ✅

File: spec/models/user_spec.rb (hunk 2 above)
Comment (jsug9) [reply spec/models/user_spec.rb]: Dear @MahmoudBakr23, sorry, we thought it wasn't required.

File: README.md (hunk 3 above)
Comment (jsug9) [reply README.md:5]: This is actually an activity for Friday, so we won't do that yet.

**Review by FrederickMih — APPROVED** [review]
Hi Team, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**